common/dynamicconfig/configstore/config_store_client.go

```diff
@@ -593,3 +606,52 @@ func convertFromDataBlob(blob *types.DataBlob) (interface{}, error) {
 		return nil, errors.New("unsupported blob encoding")
 	}
 }
+
+func validateKeyDataBlobPair(key dc.Key, blob *types.DataBlob) error {
+	value, err := convertFromDataBlob(blob)
+	if err != nil {
+		return err
+	}
+	err = fmt.Errorf("key value pair mismatch, key type: %T, value type: %T", key, value)
+	switch key.(type) {
+	case dc.IntKey:
+		if _, ok := value.(int); !ok {
+			floatVal, ok := value.(float64)
+			if !ok { // int can be decoded as float64
+				return err
+			}
+			if floatVal != math.Trunc(floatVal) {
+				return errors.New("value type is not int")
+			}
+		}
+	case dc.BoolKey:
+		if _, ok := value.(bool); !ok {
+			return err
+		}
+	case dc.FloatKey:
+		if _, ok := value.(float64); !ok {
+			return err
+		}
+	case dc.StringKey:
+		if _, ok := value.(string); !ok {
+			return err
+		}
+	case dc.DurationKey:
+		if _, ok := value.(time.Duration); !ok {
+			durationStr, ok := value.(string)
+			if !ok {
+				return err
+			}
+			if _, err = time.ParseDuration(durationStr); err != nil {
+				return errors.New("value string encoding cannot be parsed into duration")
+			}
+		}
```

Comment:
feels kinda odd to do this parse-and-validate here, and then do it again every time the value is retrieved in GetDurationValue, rather than just parsing it and using the parsed result...

... but that matches current behavior, so for this PR seems fine.  more of a "maybe we should change this later" though.